tests/quantization/ggml/test_ggml.py

```diff
@@ -694,6 +698,46 @@ def test_starcoder2_q6_k(self):
         EXPECTED_TEXT = 'def print_hello_world():\n    print("Hello World")\n\ndef print'
         self.assertEqual(tokenizer.decode(out[0], skip_special_tokens=True), EXPECTED_TEXT)
 
+    def test_mamba_weights_conversion_fp16(self):
+        original_model = AutoModelForCausalLM.from_pretrained(
+            self.mamba_original_model_id,
+            torch_dtype=torch.float16,
+        )
+
+        converted_model = AutoModelForCausalLM.from_pretrained(
+            self.mamba_model_id,
+            gguf_file=self.fp16_mamba_model_id,
+            torch_dtype=torch.float16,
+        )
+
+        converted_state_dict = converted_model.state_dict()
+        original_state_dict = original_model.state_dict()
+
+        for layer_name, original_params in original_state_dict.items():
+            if layer_name in converted_state_dict and layer_name != "lm_head.weight":
```

Comment:
We should have the same layers. If a layer is not in converted_state_dict, we should raise an error in this test instead.

Reply:
Hey @SunMarc! I added lm_head layer creation for Mamba and Starcoder2 and did some unit tests refacroting, please, take a look

Reply:
Hey @SunMarc, yes, sorry, small mess happened. No you should see latest changes.

Reply:
So if I understand, if we don't set the tensor, the lm_head won't be created or just that there is no weights ? Could you print the model to see the architecture ? We have the function tie_weights() to tie the weights of the lm_head and the embedding. Maybe that could be useful ?

Reply:
@SunMarc, No, you are right, I was confused a little bit by keya comparison between loaded model and gguf one and I saw that tie.weights() was performed on original model, but later, actually it is performed on gguf too. So, we definetely do not need to create anything explicitly. I fixed my latest changes, now, everything should be ok

Reply:
Nice !